### PR TITLE
Fix busted Darwin builds and CHIPTool Simulator support

### LINKED ISSUE #3281: Building in XCode 12 for iOS simulator targets fails
#### Problem
When I try to compile in XCode 12 for one of the iOS simulator targets (say iPhone 8), I get a link error like so:
```
ld: in /${HOME}/Library/Developer/Xcode/DerivedData/Darwin-gradnxuszkbudvafkyzpssssvewn/Build/Intermediates.noindex/CHIP.build/Debug iOS-iphonesimulator/CHIP.build/out/lib/libCHIP.a(libSetupPayload.SetupPayload.cpp.o), building for iOS Simulator, but linking in object file built for iOS, for architecture arm64
```

I checked older changesets, and before `b14e3e37d6038669cc46f849fc571b2f1e63c986` (switching XCode build to gn) the build would complete, but then XCode would complain that it couldn't run an arm64 binary in the simulator.

So it seems like going from XCode 11 to XCode 12 the simulator switched from running ARM binaries to x86_64 binaries?

 #### Proposed Solution
No idea offhand...

@rwalker-apple @bhaskar-apple @shana-apple @sagar-apple @andy31415

### LINKED ISSUE #3350: Can't compile for iOS device (not simulator)
#### Problem
Trying to build `master` for a physical iOS device, in either XCode 11 or XCode 12, fails with:
```
ld: '${HOME}/Library/Developer/Xcode/DerivedData/Darwin-gradnxuszkbudvafkyzpssssvewn/Build/Intermediates.noindex/CHIP.build/Debug iOS-iphoneos/CHIP.build/out/lib/libCHIP.a(libSetupPayload.SetupPayload.cpp.o)' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. for architecture arm64
```
Looking at the build configuration, bitcode is enabled for the "CHIP" project for "Debug iOS" and "Release iOS".  It's disabled for the "CHIPTool" project.  Enabling bitcode for "CHIPTool" does not help.  Disabling bitcode for "CHIP" lets me compile, but when running it crashes on startup in ``dyld`__abort_with_payload``.

Bisecting shows things worked before this changeset:
```
commit 8567de6b6660d147cbaa5e9da607edc7dabe5a15
Author: Justin Wood <woody@apple.com>
Date:   Mon Oct 19 10:09:31 2020 -0700

    Fixing iOS build parameters (#3299)
```

 #### Proposed Solution
No idea yet.  @woody-apple @pan-apple @bhaskar-apple @shana-apple

### PR DESCRIPTION
**Pull request by sagar-apple**
#### Problem
The Darwin builds have regressed and don't seem to have support for XCode 12. 

1. The CHIPTool App fails compilation with a bitcode failure (the cpp code isn't being compiled with bitcode support)
2. The CHIPTool App crashes on launch on a device
3. Simulator builds don't work

 #### Summary of Changes
1. Disabled bitcode for the CHIP Framework to get it compiling again (fixes problem 1)
2. Added `DYLIB_INSTALL_NAME_BASE = @rpath` to the CHIP Framework build settings so that CHIPTool finds the framework installed correctly (fixes problem 2)
3. Cleaned up the configuration names to follow "Release" "Debug" convention. 
4. Fixed the simulator builds by removing a possibly incorrect `VALID_ARCHS` entry from the CHIP Framework and CHIPTool (fixes problem 3)

 Fixes #3281 Fixes #3350

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
This fixes https://github.com/project-chip/connectedhomeip/issues/3350 too, I am guessing?

**Comment by rwalker-apple** [comment]
I have a fix for the bitcode issue...

**Review by mspang — APPROVED** [review]

**Comment by sagar-apple** [comment]
> I have a fix for the bitcode issue...
Happy to rebase if that goes in

**Comment by sagar-apple** [comment]
> This fixes #3350 too, I am guessing?

I think this is part of the fix for 3350, Rob's Bitcode fix gets you past compile and this gets you past the crash. For now this just brute forces past the compile (by disabling Bitcode)

File: src/darwin/Framework/chip_xcode_build_connector.sh [hunk 1]
```diff
@@ -78,7 +83,7 @@ declare -a args=(
     'target_cpu="'"$ARCHS"'"'
     'target_defines='"$target_defines"
     'default_configs_cosmetic=[]'
-    'target_cflags=["-target","'"$ARCHS"'-'"$LLVM_TARGET_TRIPLE_VENDOR"'-'"$LLVM_TARGET_TRIPLE_OS_VERSION"'", "-DNDEBUG"]'
+    'target_cflags=["-target","'"$ARCHS"'-'"$LLVM_TARGET_TRIPLE_VENDOR"'-'"$LLVM_TARGET_TRIPLE_OS_VERSION"'", "-DNDEBUG", '"$bitcode"']'
```
Comment (rwalker-apple) [inline src/darwin/Framework/chip_xcode_build_connector.sh:86]: glad this works

**Review by rwalker-apple — APPROVED** [review]

File: src/darwin/Framework/chip_xcode_build_connector.sh (hunk 1 above)
Comment (sagar-apple) [reply src/darwin/Framework/chip_xcode_build_connector.sh:86]: At least as far as I can tell :). Verified with the simulator and device builds (they use different Bitcode settings apparently)

**Review by andy31415 — APPROVED** [review]

**Comment by sagar-apple** [comment]
@BroderickCarlin @hawk248 @jelderton ?

**Review by saurabhst — APPROVED** [review]

File: src/darwin/Framework/CHIP.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -617,26 +371,29 @@
...
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				ARCHS = arm64;
 				CHIP_ROOT = "$(SRCROOT)/../../..";
 				CODE_SIGN_STYLE = Automatic;
 				DEFINES_MODULE = YES;
 				DEVELOPMENT_TEAM = "";
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
```
Comment (rwalker-apple) [inline src/darwin/Framework/CHIP.xcodeproj/project.pbxproj:396]: what does this do?

Comment (sagar-apple) [reply src/darwin/Framework/CHIP.xcodeproj/project.pbxproj:396]: My basic understanding is that dylib carry their "install path" information internally. When something (CHIPTool) links with CHIP, it tries to copy it over from its "install path" based on what the dylib says. For embedded fwks (most uses cases of 3rd party apps linking 3rd party frameworks) this just means "rpath" or runpath (the default runtime search path).

**Pull request merged**